dns/ddclient/src/opnsense/service/templates/OPNsense/ddclient/ddclient.conf

```diff
@@ -69,6 +66,9 @@ ssl=yes
 {%            if account.wildcard|default('0') == '1' %}
 wildcard=yes
 {%            endif %}
+{%            if account.zone %}
```

Comment:
can you try if the zone is properly cleared when editing an entry with a zone and changing it to one without one? If that's the case, I don't mind moving it here.

Reply:
I have just test it and when we change it with one without zone, the zone parameter stay ...
What I see is that it's the same for the wildcard parameter.
I'm still looking for a solution.

Reply:
I think we need to change something small in core to fix this more easily here. 
When we omit the `disabled` fields from the data that is being pushed to the endpoint [1], we should be able to reset the missing fields to their defaults (empty) in the controller [2].

1. https://github.com/opnsense/core/blob/cece0d28dee933d8edd581f804cebfaef8f66833/src/opnsense/www/js/opnsense.js#L69

2. https://github.com/opnsense/plugins/blob/bb3f06227fe48049b43a68eed3c9a94ee44044c7/dns/ddclient/src/opnsense/mvc/app/controllers/OPNsense/DynDNS/Api/AccountsController.php#L51

When fields aren't provided in the payload, they shouldn't be touched which likely makes the first change rather safe.

Reply:
What I see in this [ddclient.in](https://github.com/ddclient/ddclient/blob/develop/ddclient.in), only five provider use the "zone" parameter (ZoneEdit, NFSN, NSUpdate, CloudFlare, Gandi.net). In this case, the simplest way would not be to just use " if account.zone == ZoneEdit or account.zone == NFSN ..." for this five provider, like it was made before my proposal ?
When I made it, I didn't think it would be so complicated ^^.

Reply:
for readability it might be better to leave the setting with the provider in that case (just like it was for cloudflare), we can always improve this later.